src/builds/appveyor.rs

```diff
@@ -0,0 +1,88 @@
+use std::io::Read;
+
+use chrono::{DateTime, UTC};
+use diesel;
+use diesel::prelude::*;
+use hyper::Client;
+use hyper::header::Accept;
+use hyper::net::HttpsConnector;
+use hyper_native_tls::NativeTlsClient;
+use serde::de::DeserializeOwned;
+use serde_json;
+
+use DB_POOL;
+use domain::builds::Build;
+use error::DashResult;
+
+static ROOT_URL: &'static str = "http://ci.appveyor.com/api";
+
+pub fn get_and_insert_build(build: &str) -> DashResult<()> {
+    let url = format!("{}/projects/rust-lang/rust/build/{}", ROOT_URL, build);
+    let response: ResponseFromJson = get(&url)?;
+
+    let conn = &*DB_POOL.get()?;
+    for job in response.build.jobs.iter() {
+        if job.finished.is_none() || job.status == "cancelled" {
+            continue;
+        }
+
+        let duration = job.finished.unwrap().signed_duration_since(job.started);
+
+        let b = Build {
+            number: response.build.id,
+            builder_name: "appveyor".to_string(),
+            builder_os: "windows".to_string(),
+            builder_env: job.name.clone(),
+            successful: job.status == "success",
+            message: job.status.to_owned(),
+            duration_secs: Some(duration.num_seconds() as i32),
+            start_time: Some(job.started.naive_utc()),
+            end_time: job.finished.map(|dt| dt.naive_utc()),
```

Comment:
This makes me think that we just want to put the whole block in a typesafe `if let Some(finished) = job.finished {...}`, since here `duration_secs` is unconditionally `Some`.